src/components/lib/CopyAddressField.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useRef } from 'react'
+import QRCode from 'qrcode.react'
+
+import QRCodeIcon from '../svgs/QRCodeIcon'
+
+const CopyAddressField = ({ address, addressUrl }) => {
+  const [isCopied, setIsCopied] = useState(false)
+  const hiddenCopyFieldRef = useRef(null)
+
+  const handleCopyClick = () => {
+    hiddenCopyFieldRef.current.select()
+    document.execCommand('copy')
+    hiddenCopyFieldRef.current.blur()
+    setIsCopied(true)
+  }
+
+  return (
+    <div className="copy-address-field">
+      <div className="copy-address">
+        { addressUrl ? (
```

Comment:
As above, I'm not sure this guard is really necessary. Thoughts?

Reply:
This one is actually intentional - we only put in the QR code widget if an `addressUrl` is provided. When we use the `CopyAddressField` on the pay page, we want the QR code, but when it's used on the congrats page, we don't want to provide a QR code since it's not applicable.

Reply:
Renamed in b110333